pkg/autopilot/controller/signal/k0s/download.go

```diff
@@ -88,6 +90,7 @@ func (b downloadManifestBuilderK0s) Build(signalNode crcli.Object, signalData ap
 			ExpectedHash: signalData.Command.K0sUpdate.Sha256,
 			Hasher:       sha256.New(),
 			DownloadDir:  b.k0sBinaryDir,
+			Filename:     filepath.Join(b.k0sBinaryDir, apconst.K0sTempFilename),
```

Comment:
Hmm, shouldn't this be:

```suggestion
			Filename:     apconst.K0sTempFilename,
```

??? Otherwise, this would make the `DownloadDir` somehow superfluous.

Reply:
In grab, if the `Filename` is set, it is used as absolute path when writing the file. So I think it would be correct to actually set the `DownloadDir` as empty in this case